src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/create/workflow/CreateNewExperimentWorkflow.java

```diff
@@ -0,0 +1,304 @@
+import org.breedinginsight.brapps.importer.model.workflow.ExperimentWorkflow;
+import org.breedinginsight.brapps.importer.services.processors.experiment.ExperimentWorkflowNavigator;
+
+import javax.inject.Singleton;
+import java.util.Optional;
+
+@Slf4j
+@Getter
+@Singleton
+public class CreateNewExperimentWorkflow implements ExperimentWorkflow {
+    private final ExperimentWorkflowNavigator.Workflow workflow;
+    private final PopulateExistingPendingImportObjectsStep populateExistingPendingImportObjectsStep;
+    private final PopulateNewPendingImportObjectsStep populateNewPendingImportObjectsStep;
+    private final CommitPendingImportObjectsStep commitPendingImportObjectsStep;
+    private final ValidatePendingImportObjectsStep validatePendingImportObjectsStep;
+    private final ImportStatusService statusService;
+    private final ExperimentPhenotypeService experimentPhenotypeService;
+
+    @Inject
+    public CreateNewExperimentWorkflow(PopulateExistingPendingImportObjectsStep populateExistingPendingImportObjectsStep,
+                                       PopulateNewPendingImportObjectsStep populateNewPendingImportObjectsStep,
+                                       CommitPendingImportObjectsStep commitPendingImportObjectsStep,
+                                       ValidatePendingImportObjectsStep validatePendingImportObjectsStep,
+                                       ImportStatusService statusService,
+                                       ExperimentPhenotypeService experimentPhenotypeService) {
+        this.populateExistingPendingImportObjectsStep = populateExistingPendingImportObjectsStep;
+        this.populateNewPendingImportObjectsStep = populateNewPendingImportObjectsStep;
+        this.commitPendingImportObjectsStep = commitPendingImportObjectsStep;
+        this.validatePendingImportObjectsStep = validatePendingImportObjectsStep;
+        this.statusService = statusService;
+        this.experimentPhenotypeService = experimentPhenotypeService;
+        this.workflow = ExperimentWorkflowNavigator.Workflow.NEW_OBSERVATION;
+    }
+
+    private ImportPreviewResponse runWorkflow(ImportContext context) throws Exception {
+
+        ImportUpload upload = context.getUpload();
+        boolean commit = context.isCommit();
+        List<BrAPIImport> importRows = context.getImportRows();
+
+        // Make sure the file does not contain obs unit ids before proceeding
+        if (containsObsUnitIDs(context)) {
+            throw new HttpStatusException(HttpStatus.UNPROCESSABLE_ENTITY, "ObsUnitIDs are detected");
+        }
+
+        statusService.updateMessage(upload, "Checking existing experiment objects in brapi service and mapping data");
+
+        ProcessedPhenotypeData phenotypeData = experimentPhenotypeService.extractPhenotypes(context);
+        ProcessContext processContext = populateExistingPendingImportObjectsStep.process(context, phenotypeData);
```

Comment:
What existing pending import objects would there be when creating a new experiment?

Reply:
Could be an existing trial for example if you were appending an environment.

Reply:
Originally we were going to have a separate workflow for appending an experiment environment but we decided to keep to together to limit scope.

Reply:
I see.